langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/utils/ArrayUtils.java

```diff
@@ -101,26 +102,31 @@ public static BError createOpNotSupportedError(Type type, String op) {
     public static BArray createEmptyArrayFromTuple(BArray arr) {
         Type arrType = TypeUtils.getImpliedType(arr.getType());
         TupleType tupleType = (TupleType) arrType;
-        List<Type> memTypes = new ArrayList<>();
         List<Type> tupleTypes = tupleType.getTupleTypes();
+        Type restType = tupleType.getRestType();
         boolean isSameType = true;
         Type sameType = null;
         if (!tupleTypes.isEmpty()) {
             sameType = tupleTypes.get(0);
-            memTypes.add(sameType);
-        }
-        for (int i = 1; i < tupleTypes.size(); i++) {
-            isSameType &= sameType == tupleTypes.get(i);
-            memTypes.add(tupleTypes.get(i));
+            for (int i = 1; i < tupleTypes.size(); i++) {
+                if (tupleTypes.get(i) != sameType) {
+                    isSameType = false;
+                    break;
+                }
+            }
         }
-        Type restType = tupleType.getRestType();
+        List<Type> memTypes = new ArrayList<>(tupleTypes);
         // If there's a tuple-rest-descriptor the array will not be of the same type even if other types are the same
         if (restType != null) {
             isSameType = false;
             memTypes.add(restType);
```

Comment:
This is wrong, we can have tuple types like this `[float, float, float...] tup = [1,2,3,4];` and we can create `float[]` arrays for them.

Reply:
Addressed.